src/arena.rs

```diff
@@ -58,8 +54,10 @@ impl<T> Handle<T> {
         }
     }
 
+    /// Returns the zero-based index of this handle.
     pub fn index(self) -> usize {
-        self.index as usize
+        let index = self.index.get() - 1;
```

Comment:
would it be simpler to just start an `Arena` with on dummy element all the time?

Reply:
It could work, but since `Arena` is generic I'm not sure what value to give to the dummy element?

Reply:
could require `T: Default`

Reply:
I've tried, started adding `derive(Default)` to various types in this crate, but ended up requiring that some `spirv_headers` types implement `Default`.

Reply:
If it's too difficult, it may be ok to just land what you had originally.
Technically speaking, you can always `impl Default` manually for our structs in order to be able to use that approach. Please make a judgement call for whether there is too much boilerplace, and we'll go from there.